File: Python files/CO2_loss_script_temp.py
#author: Riley Doyle
#date: 7/28/20
#file: CO2_loss_script_temp
#status: working 

#clear all
from IPython import get_ipython
get_ipython().magic('reset -sf')

from calc_Ks import *
from calc_alphas import *
from calc_CO2_loss import *
from calc_density import *
import numpy as np
import matplotlib.pyplot as plt
from calc_CO2_loss_temp import *

#figures with different alkalinities
T = 10 + 273.15
while T <= 30 + 273.15:
    S = 35
    Tc = T - 273.15; #celcius
    P = 10; #(dbar)
    t = Tc*1.00024;
    p = P/10;
    den = calc_density(S, t, p); #(kg/m3)
    K1 = calc_K1(T,S)*(den/1000) #mol/L
    pK1 = -np.log10(K1)
    K2 = calc_K2(T,S)*(den/1000) #mol/L
    pK2 = -np.log10(K2)
    Kh = calc_Kh(T,S)*(den/1000) #mol/L/atm
    PCO2 = 0.000416
    alkin = 2
    alkend = 27
    delalk = 5
    pHin = 6
    pHend = 8.2
    delpH = 0.1
    d = 0.15
    kLa = 0.5*(1.024**((T-273.15)-20))
    y = calc_CO2_loss(pK1, pK2, Kh, kLa, d, PCO2, pHin, pHend, delpH, alkin, alkend, delalk)
    plt.xlabel('pH')
    plt.ylabel('CO$_2$ loss to the atmosphere (g m$^{-2}$ day$^{-1})$')
    plt.legend(['alk = 2 meq/L', 'alk = 7 meq/L', 'alk = 12 meq/L', 'alk = 17 meq/L', 'alk = 22 meq/L'], frameon=False)
    plt.figure()
    T += 10

#figure with different temperatures
Tin = 10 + 273.15
Tend = 70 + 273.15
delT = 20
S = 35
PCO2 = 0.000416
alk = 2.5
pHin = 6
pHend = 8.5
delpH = 0.1
d = 0.15
y =calc_CO2_loss_temp (PCO2, alk, d, pHin, pHend, delpH, S, Tin, Tend, delT)
plt.xlabel('pH')
plt.ylabel('CO$_2$ loss to the atmosphere (g m$^{-2}$ day$^{-1})$')
plt.legend(['T= 10 $^o$C', 'T= 30 $^o$C', 'T= 50 $^o$C'])
plt.axis([6, 8.3,0,200])
plt.show()

File: Matlab files/CO2_loss_script_temp.m
%Author: Riley Doyle
%Date: 7/31/20
%Dependent functions: calc_K1, calc_K2, calc_CO2_loss_temp
%Inputs: S, CO2sat, kLa, pH, alk
%Outputs:CO2 loss to atmosphere without algae growth

%delete all figures and variables in the workspace
clear 
close all


%define variables
Tin = 10 + 273.15; %temp in Kelvins
Tend = 30 + 273.15;
delT = 10;
S = 35; %(salinity in g/kg)
CO2sat = 0.012716352; %(mole/m3) saturation concentration of CO2 in water
kLa = 0.5; %(1/hr)
pHin = 6.5; %no units
pHend = 8.5; %no units
delpH = 0.1; %no units
d = .15; %(m) depth of pond
alk = 2.5;

%define r_temp
r_temp = calc_CO2_loss_temp (S, CO2sat, alk, kLa, pHin, pHend, delpH, Tend, Tin, delT);

%modify r for plotting
%x axis is the first column of r
x = r_temp(:,1);
r_temp(:,1) = [];
r = r_temp*d; %(g/m2*day)
L = {'-', '--', '-.'};

%plot CO2 loss vs pH for multiple alkalinities
hold on
plot(x,r(:,1), 'Linestyle', L{1})
plot(x,r(:,2), 'Linestyle', L{2})
plot(x,r(:,3), 'Linestyle', L{3})

xlabel('pH')
ylabel('CO_2 loss to the atmosphere (g m^{-2} day^{-1})')
legend('T = 10 ^oC','T = 20 ^oC','T = 30 ^oC')